cmd/cloud.go

```diff
@@ -251,14 +251,16 @@ This will execute the test on the k6 cloud service. Use "k6 login cloud" to auth
 			)
 
 			progressCtx, progressCancel := context.WithCancel(globalCtx)
-			progressBarWG := &sync.WaitGroup{}
-			progressBarWG.Add(1)
-			defer progressBarWG.Wait()
 			defer progressCancel()
-			go func() {
-				showProgress(progressCtx, conf, []*pb.ProgressBar{progressBar}, logger)
-				progressBarWG.Done()
-			}()
+			progressBarWG := &sync.WaitGroup{}
+			if !quiet {
+				progressBarWG.Add(1)
+				defer progressBarWG.Wait()
+				go func() {
+					showProgress(progressCtx, conf, []*pb.ProgressBar{progressBar}, logger)
+					progressBarWG.Done()
+				}()
+			}
```

Comment:
The problem is that:
1. `--quiet` just makes it way more likely (and possible at all) to see the race, as writing to stdout is *synchronous* so it basically makes the synchronization for us .... or just slows k6 down enough that it doesn't happen :man_shrugging: 
2. The problem actually (as somewhat, but not very well(IMO) shown by the stack trace) is that [GetProgress](https://github.com/k6io/k6/blob/master/lib/executor/base_executor.go#L75) is defined on the * non-pointer receiver* which means that when it's invoked it will *copy* the struct effectively *read*ing all of its fields. At the same time the `Init` *write*s to one of the fields which makes the race listed above.

So we should probably have all the `BaseExecutor` methods be on the pointer receiver as it won't copy the struct.

Reply:
Resolved by f9e0ab65. Thanks! :clap: